From 82d13e677394dbe225047f60f7d6207c2009e20a Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 25 Apr 2023 21:01:38 +0100
Subject: [PATCH 01/14] gnu: python-pytest-doctestplus: Update to 0.12.1.

* gnu/packages/python-check.scm (python-pytest-doctestplus): Update to
0.12.1.
[build-system]: Use pyproject-build-system.
[arguments]<phases>: Remove modify-phases and add test-flags option
instead.
[native-inputs]: Add python-numpy.
---
 gnu/packages/python-check.scm | 31 ++++++++++++-------------------
 1 file changed, 12 insertions(+), 19 deletions(-)

diff --git a/gnu/packages/python-check.scm b/gnu/packages/python-check.scm
index c42cd426b1..dd0600a2fc 100644
--- a/gnu/packages/python-check.scm
+++ b/gnu/packages/python-check.scm
@@ -577,32 +577,25 @@ (define-public python-pytest-arraydiff
 (define-public python-pytest-doctestplus
   (package
     (name "python-pytest-doctestplus")
-    (version "0.11.2")
+    (version "0.12.1")
     (source
      (origin
        (method url-fetch)
        (uri (pypi-uri "pytest-doctestplus" version))
        (sha256
-        (base32 "0j1lvlj3ps975q9hmg8i6rpqm0313j3r18bc3l8mz6khb7vav4zk"))))
-    (build-system python-build-system)
+        (base32 "10ciqylgziihxwxryxvxgmkqgws51pqcarn0gbh1d4cxx55rx5vs"))))
+    (build-system pyproject-build-system)
     (arguments
-     '(#:phases
-       (modify-phases %standard-phases
-         (replace 'check
-           (lambda* (#:key inputs outputs tests? #:allow-other-keys)
-             ;; Make the installed plugin discoverable by Pytest.
-             (when tests?
-               (add-installed-pythonpath inputs outputs)
-               (invoke "python" "-m" "pytest" "-k"
-                       (string-append   ; skip tests that require remote data
-                        "not test_remote_data_url"
-                        " and not test_remote_data_float_cmp"
-                        " and not test_remote_data_ignore_whitespace"
-                        " and not test_remote_data_ellipsis"
-                        " and not test_remote_data_requires"
-                        " and not test_remote_data_ignore_warnings"))))))))
+     (list #:test-flags
+           #~(list "-k" (string-append
+                         "not test_remote_data_url"
+                         " and not test_remote_data_float_cmp"
+                         " and not test_remote_data_ignore_whitespace"
+                         " and not test_remote_data_ellipsis"
+                         " and not test_remote_data_requires"
+                         " and not test_remote_data_ignore_warnings"))))
     (native-inputs
-     (list python-pytest python-setuptools-scm))
+     (list python-numpy python-pytest python-setuptools-scm))
     (home-page "https://github.com/astropy/pytest-doctestplus")
     (synopsis "Pytest plugin with advanced doctest features")
     (description
-- 
2.39.2

